x-pack/metricbeat/module/openai/usage/config.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package usage
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+type Config struct {
+	APIKeys    []apiKeyConfig   `config:"api_keys" validate:"required"`
+	APIURL     string           `config:"api_url" validate:"required"`
+	Headers    []string         `config:"headers"`
+	RateLimit  *rateLimitConfig `config:"rate_limit"`
+	Timeout    time.Duration    `config:"timeout" validate:"required"`
+	Collection collectionConfig `config:"collection"`
+}
+
+type rateLimitConfig struct {
+	Limit *int `config:"limit" validate:"required"`
+	Burst *int `config:"burst" validate:"required"`
+}
+
+type apiKeyConfig struct {
+	Key string `config:"key" validate:"required"`
+}
+
+type collectionConfig struct {
+	LookbackDays int  `config:"lookback_days" validate:"required"`
+	Realtime     bool `config:"realtime"`
+}
+
+func defaultConfig() Config {
+	return Config{
+		APIURL:  "https://api.openai.com/v1/usage",
+		Timeout: 30 * time.Second,
+		RateLimit: &rateLimitConfig{
+			Limit: ptr(60),
```

Comment:
What defines these limits ?
If you could point me to documentation due to which we have chosen these defaults ?

Reply:
The time is a sane default. And the rate-limit is decided as per testing. If you see the meta ticket, you'd find an issue there where we have researched about rate-limits of this API. Not linking it here as it's private. Can share over Slack.